nixos/modules/services/network-filesystems/ceph.nix

```diff
@@ -401,7 +400,13 @@ in
         mkMerge targets;
 
     systemd.tmpfiles.rules = [
+      "d /etc/ceph - ceph ceph - -"
       "d /run/ceph 0770 ceph ceph -"
-    ];
+      "d /var/lib/ceph - ceph ceph - -"]
+    ++ optionals cfg.mgr.enable [ "d /var/lib/ceph/mgr - ceph ceph - -"]
+    ++ optionals cfg.mon.enable [ "d /var/lib/ceph/mon - ceph ceph - -"]
+    ++ optionals cfg.osd.enable [ "d /var/lib/ceph/osd - ceph ceph - -"]
+    ++ (map (daemonName: "d /var/lib/ceph/mgr/${cfg.global.clusterName}-${daemonName} - ceph ceph - -") cfg.mgr.daemons)
+    ++ (map (daemonName: "d /var/lib/ceph/osd/${cfg.global.clusterName}-${daemonName} - ceph ceph - -") cfg.osd.daemons);
```

Comment:
You're missing a map for the mon daemons, they follow the same pattern as the other daemons.

Reply:
These are somewhat redundant with the `StateDirectory` directives in the individual units, which should be created by systemd when starting the units.

However, as some work needs to be done to initialize the cluster before these units can be started, just relying on `StateDirectory` isn't enough, as the initialization tools would fail otherwise.

I didn't want to duplicate everything from `StateDirectory` - happy for better suggestions w.r.t. cluster initialization.

Reply:
Well, the mon daemons are the most important daemons in the cluster. They are the first that need to be initialized in a cluster, so I think that we sadly need to duplicate everything from StateDirectory in tmpfiles too.